saleor/core/jwt.py

```diff
@@ -103,6 +108,49 @@ def get_user_from_payload(payload: Dict[str, Any]) -> Optional[User]:
 
 def get_user_from_access_token(token: str) -> Optional[User]:
     payload = jwt_decode(token)
-    if payload["type"] != JWT_ACCESS_TYPE:
+    if payload["type"] not in [JWT_ACCESS_TYPE, JWT_THIRDPARTY_ACCESS_TYPE]:
         return None
-    return get_user_from_payload(payload)
+    permission_limits = payload.get(PERMISSION_LIMITS_FIELD, None)
+    user = get_user_from_payload(payload)
+    if permission_limits is not None:
+        user.permission_limits = permission_limits  # type: ignore
+    return user
+
+
+def create_access_token_for_app(app: "App", user: "User"):
+    """Create access token for app.
+
+    App can use user jwt token to proceed given operation on the Saleor side.
+    The token which can be used by App has additional field defining the permissions
+    assigned to it. The permissions set is the intersection of user permissions and
+    app permissions.
+    """
+
+    permissions_dict = {
+        enum.codename: enum.name
+        for permission_enum in PERMISSIONS_ENUMS
+        for enum in permission_enum
+    }
+    app_permissions = app.permissions.all()
+    app_permission_enums = {permissions_dict[perm.codename] for perm in app_permissions}
+
+    if user.is_superuser:
+        user_permissions = get_permissions()
```

Comment:
When I inspecting the coverage report I found that we don't have any test for this line. Can you add a test for this function as a superuser.